storage/perfschema/pfs_setup_object.cc

```diff
@@ -175,15 +177,15 @@ int insert_setup_object(enum_object_type object_type, const String *schema,
   if (unlikely(pins == NULL))
     return HA_ERR_OUT_OF_MEM;
 
-  static uint PFS_ALIGNED setup_object_monotonic_index= 0;
+  static std::atomic<uint> setup_object_monotonic_index(0);
```

Comment:
PFS_ALIGNED is needed as well.

Reply:
I also dropped `PFS_ALIGNED` in other files such as `pfs_instr.cc` and `pfs_diget.cc` (static or not), presumably I should also add them back?

Reply:
If original the intention of the `PFS_ALIGNED` around `digest_monotonic_index` was to prevent false sharing of cache lines, then also the following global variable that is expected to be part of the false sharing should be declared as `PFS_ALIGNED`. It could be better to declare a `struct` around it, so that the `struct` itself will be padded to a multiple of the `PFS_ALIGNED` size. It would not seem to me that `digest_full` could be part of false sharing; it is often being changed together with that field. Something like the following might work:
```c++
struct
{
  PFS_ALIGNED Atomic_counter<uint32_t> monotonic_index;
  bool full;
} digest;
```
Note: Global or `static` variables are zero-initialized by default. An initializer like `bool digest_full=false;` is redundant.

Reply:
Hey all, sorry for the late response. Since the PR is a bit stale so I rebased my changes to 10.5. In 10.5 these atomic counters are already wrapped in a structure (`struct PFS_cacheline_uint32` for example) so it can be simply done by replacing the member with atomic counter part (`uint32 m_u32;` to `std::atomic<uint32> m_u32;`)